extractor/image/image_test.go

```diff
@@ -118,6 +118,12 @@ func TestNewDockerArchiveImage(t *testing.T) {
 				fileName: "../testdata/alpine-310.tar.gz",
 			},
 		},
+		{
+			name: "happy path with OCI Image Format",
+			args: args{
+				fileName: "../testdata/test.oci",
+			},
+		},
```

Comment:
Could you also add some sad paths?

Reply:
We already have a sad path. Isn't it enough?
https://github.com/aquasecurity/fanal/blob/387cf0c049fd5e9073ba760125964f0f3b81bfc9/extractor/image/image_test.go#L127-L133

Reply:
I feel it's not. There are still other ways the function can fail. See the coverage as shown below. Red indicates no coverage.
<img width="520" alt="image" src="https://user-images.githubusercontent.com/1254783/79898990-f411fb00-83c0-11ea-9b88-8e4506024c2e.png">

Reply:
Do you think we should go through all paths? Should I add 4 more cases?

Reply:
I tried to add tests but I noticed another issue. Currently all errors are masked upstream here: https://github.com/aquasecurity/fanal/pull/101/files#diff-525dd35201b72b43ad63b85dd2720ba6R75-R80 into a single error of `"invalid archive"`.

So we will never be able to see the underlying error that happened in `tryOCI` func. This would be an example to test our return types in such a case. 

I agree that the library is well tested but it doesn't guard against regressions that happens because of future refactorings of `tryOCI` func or in this case where the error isn't what we expected it to be.

Reply:
I didn't have a way to handle multiple errors, so I introduced hashicorp/go-multierror.
https://github.com/aquasecurity/fanal/pull/101/commits/12c9696ce44781570891ea24e76d2146159e9ebb

If you feel we should add more tests, please point it out. Of course, you can add it yourself.

Reply:
Cool! I added a few more tests: https://github.com/aquasecurity/fanal/pull/101/commits/98fd6e4bb85627e05c733560cfd684b8fa650003